docs/reference/modules/cluster/disk_allocator.asciidoc

```diff
@@ -37,23 +37,23 @@ file or updated dynamically on a live cluster with the
     changed in 8.0. This setting can be set to `true` to enable the
     disk watermarks for a single data node cluster (will become default in 8.0).
 
+[[cluster-routing-flood_stage]]
 `cluster.routing.allocation.disk.watermark.flood_stage`::
 +
 --
-Controls the flood stage watermark. It defaults to 95%, meaning that
-Elasticsearch enforces a read-only index block
+Controls the flood stage watermark, which defaults to 95%. Elasticsearch enforces a read-only index block
 (`index.blocks.read_only_allow_delete`) on every index that has one or more
-shards allocated on the node that has at least one disk exceeding the flood
-stage. This is a last resort to prevent nodes from running out of disk space.
-The index block is automatically released once the disk utilization falls below
+shards allocated on the node, and that has at least one disk exceeding the flood
+stage. This setting is a last resort to prevent nodes from running out of disk space.
+The index block is automatically released when the disk utilization falls below
 the high watermark.
 
-NOTE: You can not mix the usage of percentage values and byte values within
-these settings. Either all are set to percentage values, or all are set to byte
-values. This is so that we can we validate that the settings are internally
-consistent (that is, the low disk threshold is not more than the high disk
-threshold, and the high disk threshold is not more than the flood stage
-threshold).
+NOTE: You cannot mix the usage of percentage values and byte values within
+these settings. Either all values are set to percentage values, or all are set to byte
+values. This enforcement is so that Elasticsearch can validate that the settings are internally
+consistent, ensuring that the low disk threshold is less than the high disk
+threshold, and the high disk threshold is less than the flood stage
+threshold. Therefore, you typically do not need to modify this setting.
```

Comment:
I don't think the last sentence follows from the description. I think it'd be lovely to have a paragraph on when it is useful to modify these settings, but that it'd belong more at the top of the page than here. I think if you modify the settings you typically modify all of them. I *think* you mostly are interested in modifying them if you have a very big disk and 85%/90%/95% would be quite wasteful. Of if you write *lots* of data very quickly compared to the amount of disk so the defaults aren't conservative enough.

Reply:
I removed the last sentence [this commit](https://github.com/elastic/elasticsearch/pull/57184/commits/f82d51a552c8988ad4f4053ca3ed0914845bf126). If you think an expanded explanation of why users might want to modify these settings is warranted, let me know.

Reply:
I do think it'd be nice, but this is a good change as it stands now anyway. Merge anytime you are ready!